### PR TITLE
Added shader reflection part to source-only PDB.

### PR DESCRIPTION
**Pull request by adam-yang**

### CODE REVIEW

File: tools/clang/unittests/HLSL/CompilerTest.cpp [hunk 1]
```diff
@@ -1213,6 +1215,42 @@ static void VerifyPdbUtil(dxc::DxcDllSupport &dllSupport,
     TestArgumentPair(Args, ExpectedArgs);
   }
 
+  // Shader reflection
+  if (TestReflection) {
+    CComPtr<IDxcContainerReflection> pContainerRef;
+    VERIFY_SUCCEEDED(dllSupport.CreateInstance(CLSID_DxcContainerReflection, &pContainerRef));
+    VERIFY_SUCCEEDED(pContainerRef->Load(pBlob));
```
Comment (tex3d) [inline tools/clang/unittests/HLSL/CompilerTest.cpp:1222]: We should make `IDxcUtils::CreateReflection` work with PDB as well as with the DxilContainer from the PDB if possible.

File: tools/clang/unittests/HLSL/CompilerTest.cpp [hunk 2]
```diff
@@ -1450,6 +1498,7 @@ void CompilerTest::TestPdbUtils(bool bSlim, bool bSourceInDebugModule, bool bStr
       /*HasVersion*/ false,
       /*IsFullPDB*/  true,
       /*hasHasAndPDBName*/false,
+      /*TestReflection*/false,
```
Comment (tex3d) [inline tools/clang/unittests/HLSL/CompilerTest.cpp:1501]: Why false in this case?  Perhaps comment could help.

File: tools/clang/unittests/HLSL/CompilerTest.cpp [hunk 3]
```diff
@@ -1472,6 +1522,7 @@ void CompilerTest::TestPdbUtils(bool bSlim, bool bSourceInDebugModule, bool bStr
       /*HasVersion*/ false,
       /*IsFullPDB*/ true,
       /*hasHasAndPDBName*/true,
+      /*TestReflection*/false,
```
Comment (tex3d) [inline tools/clang/unittests/HLSL/CompilerTest.cpp:1525]: And why false here too?  By the way, typo in previous line comment: `hasHasAndPDBName`

File: tools/clang/unittests/HLSL/CompilerTest.cpp (hunk 1 above)
Comment (adam-yang) [reply tools/clang/unittests/HLSL/CompilerTest.cpp]: Done.

File: tools/clang/unittests/HLSL/CompilerTest.cpp (hunk 2 above)
Comment (adam-yang) [reply tools/clang/unittests/HLSL/CompilerTest.cpp]: Added a comment.

File: tools/clang/unittests/HLSL/CompilerTest.cpp (hunk 3 above)
Comment (adam-yang) [reply tools/clang/unittests/HLSL/CompilerTest.cpp]: Fixed.

File: tools/clang/unittests/HLSL/CompilerTest.cpp [hunk 4]
```diff
@@ -1497,8 +1495,8 @@ void CompilerTest::TestPdbUtils(bool bSlim, bool bSourceInDebugModule, bool bStr
       pCompiler,
       /*HasVersion*/ false,
       /*IsFullPDB*/  true,
-      /*hasHasAndPDBName*/false,
-      /*TestReflection*/false,
+      /*HasHashAndPdbName*/false,
+      /*TestReflection*/false, // Reflection creation interface doesn't support just
```
Comment (tex3d) [inline tools/clang/unittests/HLSL/CompilerTest.cpp:1499]: just what?

**Review by tex3d — APPROVED** [review]
Looks good.  One comment seems incomplete, but not blocking merge.

**Pull request merged**